test/zmumu_pp5tev/tp_from_aod_Data.py

```diff
@@ -0,0 +1,629 @@
+        ## track matching variables
+        tk_deltaR     = cms.InputTag("staToTkMatch","deltaR"),
+        tk_deltaEta   = cms.InputTag("staToTkMatch","deltaEta"),
+        tk_deltaR_NoZ   = cms.InputTag("staToTkMatchNoZ","deltaR"),
+        tk_deltaEta_NoZ = cms.InputTag("staToTkMatchNoZ","deltaEta"),
+    ),
+    flags = cms.PSet(
+        outerValidHits = cms.string("outerTrack.numberOfValidHits > 0"),
+        TM  = cms.string("isTrackerMuon"),
+        Glb = cms.string("isGlobalMuon"),
+        Tk  = cms.string("track.isNonnull"),
+        StaTkSameCharge = cms.string("outerTrack.isNonnull && innerTrack.isNonnull && (outerTrack.charge == innerTrack.charge)"),
+    ),
+    tagVariables = cms.PSet(
+        pt = cms.string("pt"),
+        eta = cms.string("eta"),
+        phi = cms.string("phi"),
+        nVertices = cms.InputTag("nverticesModule"),
+        combRelIso = cms.string("(isolationR03.emEt + isolationR03.hadEt + isolationR03.sumPt)/pt"),
+        chargedHadIso04 = cms.string("pfIsolationR04().sumChargedHadronPt"),
+        neutralHadIso04 = cms.string("pfIsolationR04().sumNeutralHadronEt"),
+        photonIso04 = cms.string("pfIsolationR04().sumPhotonEt"),
+        combRelIsoPF04dBeta = IsolationVariables.combRelIsoPF04dBeta,
+        l1rate = cms.InputTag("l1rate"),
+        bx     = cms.InputTag("l1rate","bx"),
+        instLumi = cms.InputTag("addEventInfo", "instLumi"),
+    ),
+    pairVariables = cms.PSet(
+        nJets30 = cms.InputTag("njets30ModuleSta"),
+        dz      = cms.string("daughter(0).vz - daughter(1).vz"),
+        pt      = cms.string("pt"), 
+        rapidity = cms.string("rapidity"),
+        deltaR   = cms.string("deltaR(daughter(0).eta, daughter(0).phi, daughter(1).eta, daughter(1).phi)"), 
+    ),
+    pairFlags = cms.PSet(),
+)
+process.njets30ModuleSta = process.njets30Module.clone(pairs = "tpPairsSta")
+
+process.tnpSimpleSequenceSta = cms.Sequence(
+    process.tagMuons +
+    process.oneTag     +
+    process.probeMuonsSta   +
+    process.tpPairsSta      +
+    process.onePairSta      +
+    process.nverticesModule +
+    process.staToTkMatchSequenceZ +
+    process.njets30ModuleSta +
+    process.addEventInfo +
+    process.l1rate +
+    process.tpTreeSta
+)
+
+## Add extra RECO-level info
+if False:
+    process.tnpSimpleSequenceSta.replace(process.tpTreeSta, process.tkClusterInfo+process.tpTreeSta)
+    process.tpTreeSta.tagVariables.nClustersStrip = cms.InputTag("tkClusterInfo","siStripClusterCount")
+    process.tpTreeSta.tagVariables.nClustersPixel = cms.InputTag("tkClusterInfo","siPixelClusterCount")
+    process.tnpSimpleSequenceSta.replace(process.tpTreeSta, process.tkLogErrors+process.tpTreeSta)
+    process.tpTreeSta.tagVariables.nLogErrFirst = cms.InputTag("tkLogErrors","firstStep")
+    process.tpTreeSta.tagVariables.nLogErrPix   = cms.InputTag("tkLogErrors","pixelSteps")
+    process.tpTreeSta.tagVariables.nLogErrAny   = cms.InputTag("tkLogErrors","anyStep")
+
+if True: 
```

Comment:
Do we need this? (This probably takes a bit of time to process)

Reply:
i guess not needed